### PR TITLE
Adding API documentation

### PR DESCRIPTION
**Pull request by jtaubensee**
## Description

**Work in progress**

* Turned on XML documentation
* Added all basic documentation
* Added empty tag <summary></summary> to all protected members that aren't designed to be extended by consumers of this library. Since these are easily searchable, we can always add this documentation.

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](./CONTRIBUTING.md).**
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR.
- [x] The pull request does not introduce breaking changes (unless a major version change occurs in the assembly and module).
- [x] If applicable, the public code is properly documented.
- [x] Pull request includes test coverage for the included changes.
- [x] The code builds without any errors.

### CODE REVIEW

File: src/Microsoft.Azure.EventHubs/EventHubClient.cs [hunk 1]
```diff
@@ -73,6 +82,10 @@ static EventHubClient Create(EventHubsConnectionStringBuilder csb)
             return eventHubClient;
         }
 
+        /// <summary>
+        /// 
+        /// </summary>
```
Comment (serkantkaraca) [inline src/Microsoft.Azure.EventHubs/EventHubClient.cs]: Summary missing.

File: src/Microsoft.Azure.EventHubs/EventHubPartitionRuntimeInformation.cs [hunk 2]
```diff
@@ -5,6 +5,9 @@ namespace Microsoft.Azure.EventHubs
 {
     using System;
 
+    /// <summary>
+    /// Contains information regarding Event Hubs partitions.
```
Comment (serkantkaraca) [inline src/Microsoft.Azure.EventHubs/EventHubPartitionRuntimeInformation.cs]: >vent Hubs partitions. [](start = 40, length = 21)

This is a single partition actually.

File: src/Microsoft.Azure.EventHubs/Primitives/AsyncLock.cs [hunk 3]
```diff
@@ -8,23 +8,39 @@ namespace Microsoft.Azure.EventHubs
     using System.Threading.Tasks;
 
     // Code based on http://blogs.msdn.com/b/pfxteam/archive/2012/02/12/10266988.aspx
+
+    /// <summary>
+    /// Used as an asynchronous semaphore for internal Event Hubs operations.
+    /// </summary>
```
Comment (serkantkaraca) [inline src/Microsoft.Azure.EventHubs/Primitives/AsyncLock.cs]: Do we need to keep this public? This is all internal.

File: src/Microsoft.Azure.EventHubs.Processor/PartitionContext.cs [hunk 4]
```diff
@@ -21,12 +24,24 @@ internal PartitionContext(EventProcessorHost host, string partitionId, string ev
...
         public string ConsumerGroupName { get; }
 
+        /// <summary>
+        /// Gets or sets the path of the event hub.
+        /// </summary>
         public string EventHubPath { get; }
 
+        /// <summary>
+        /// Gets the partition ID for the context.
+        /// </summary>
```
Comment (serkantkaraca) [inline src/Microsoft.Azure.EventHubs.Processor/PartitionContext.cs]: All 3 is getter only. Descriptions say "gets or sets"

**Comment by jtaubensee** [comment]
@serkant - I've addressed the feedback. Will you take another look?

**Pull request merged**